ksql-engine/src/main/java/io/confluent/ksql/KsqlEngine.java

```diff
@@ -481,37 +496,20 @@ public DDLCommandExec getDDLCommandExec() {
     return ddlCommandExec;
   }
 
-  @Override
   public boolean terminateQuery(final QueryId queryId, final boolean closeStreams) {
-    QueryMetadata queryMetadata = persistentQueries.remove(queryId);
-    if (queryMetadata == null) {
+    PersistentQueryMetadata persistentQueryMetadata = persistentQueries.remove(queryId);
+    if (persistentQueryMetadata == null) {
       return false;
     }
-    livePersistentQueries.remove(queryMetadata);
-    allLiveQueries.remove(queryMetadata);
+    livePersistentQueries.remove(persistentQueryMetadata);
+    allLiveQueries.remove(persistentQueryMetadata);
     if (closeStreams) {
-      queryMetadata.close();
+      persistentQueryMetadata.close();
+      new StreamsTopologyUtil()
```

Comment:
This is a utility class, so should not be instantiated.

Reply:
Or maybe we don't need it. i.e., the set of internal topics belongs to the query. So maybe the `QueryMetadata` could have a method something like `internalTopicNames(final Collection<String> topics /*this is the data from schemaRegistryClient.getAllSubjects()*/`
The `cleanupInternalTopicAvroSchemas()` method becomes part of `KsqlEngine` and we delete the util class